pycaret/internal/ensemble.py

```diff
@@ -0,0 +1,127 @@
+# Module: internal.ensemble
+# Author: Miguel Trejo
+# License: MIT
+
+# Provides an Ensemble Forecaster supporting voting, mean and median methods.
+
+# This Ensemble is only to be used internally.
+
+import pandas as pd
+import numpy as np
+import warnings
+from sktime.forecasting.base._base import DEFAULT_ALPHA
+from sktime.forecasting.base._meta import _HeterogenousEnsembleForecaster
+from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
+
+_ENSEMBLE_METHODS = ["voting", "mean", "median"]
+
+class _EnsembleForecasterWithVoting(
```

Comment:
Hi @TremaMiguel , I am wondering if you should submit this as a PR to the `sktime` repository directly instead of recreating this in PyCaret?

Reply:
We should do both IMO, it may take a while for any PRs to be accepted - we can always just remove it from here later

Reply:
Not sure if this approach is what sktime is looking for, they already have a [_PredictionWeightedEnsembler](https://github.com/alan-turing-institute/sktime/blob/master/sktime/forecasting/online_learning/_prediction_weighted_ensembler.py#L7) class for the Online Learning module, but definitively we can discuss this with them.